File: AntiClown.Web.Front/src/Dto/F1Predictions/F1PredictionDto.ts
import { F1DnfPredictionDto } from "./F1DnfPredictionDto";
import { F1SafetyCarPredictionDto } from "./F1SafetyCarsPredictionDto";

export interface F1PredictionDto {
  raceId: string;
  userId: string;
  tenthPlacePickedDriver: string;
  safetyCarsPrediction: F1SafetyCarPredictionDto;
  teamsPickedDrivers?: string[];
  firstPlaceLeadPrediction: number;
  dnfPrediction: F1DnfPredictionDto;
  driverPositionPrediction: number;
}

File: src/EntertainmentApi/AntiClown.Entertainment.Api.Dto/F1Predictions/F1PredictionDto.cs
namespace AntiClown.Entertainment.Api.Dto.F1Predictions;

public class F1PredictionDto
{
    public Guid RaceId { get; set; }
    public Guid UserId { get; set; }
    public string TenthPlacePickedDriver { get; set; }
    public F1DnfPredictionDto DnfPrediction { get; set; }
    public F1SafetyCarsPredictionDto SafetyCarsPrediction { get; set; }
    [Obsolete("2024-2025")] public string[] TeamsPickedDrivers { get; set; }
    public decimal FirstPlaceLeadPrediction { get; set; }
    public int DriverPositionPrediction { get; set; }
}